checkov/common/output/record.py

```diff
@@ -197,11 +197,12 @@ def to_string(self, compact: bool = False, use_bc_ids: bool = False) -> str:
         secret_validation_status_string = ""  # nosec
         if self.check_result["result"] == CheckResult.FAILED and \
                 hasattr(self, 'validation_status') and \
-                os.getenv("CKV_VALIDATE_SECRETS"):
-            secret_validation_status_string = f'\tStatus: {self.validation_status}\n'
+                os.getenv("CKV_VALIDATE_SECRETS") and \
+                hasattr(self, '_get_secret_validation_status_message'):  # for typing purposes, can't check with isinstance cause of circular dependency
+            secret_validation_status_string = self._get_secret_validation_status_message()
 
         if self.check_result["result"] == CheckResult.FAILED and code_lines and not compact:
-            return f"{check_message}{status_message}{severity_message}{detail}{file_details}{secret_validation_status_string}{caller_file_details}{guideline_message}{code_lines}{evaluation_message}"
+            return f"{check_message}{status_message}{secret_validation_status_string}{severity_message}{detail}{file_details}{caller_file_details}{guideline_message}{code_lines}{evaluation_message}"
```

Comment:
A small general question here :)
the parts of the validation status of the secret are not related to all other records. I am not sure, but don't we prefer to change the ending of this method, so the main record class will not even know there is a `_get_secret_validation_status_message` method? 
and the `secrets_record` as the subclass will add the logic it needs?

I couldn't find a reason why this could be a bad idea so thought of asking. WDYT? :)

Reply:
definitely would be more clean, but needs some adjustments, like adding an extra method, which is called to add this validation string and in the base calls it just returns an empty string.

Reply:
Definitely a good idea.

I could just override `to_string` by copy-pasting it from Record, but that will result in a lot of code duplication.

Definitely, there's room for better abstractions here, but I think it should be handled separately since it won't be a small change.
I think we can start by copy-pasting `to_string` to `secrets_record` and I'll work on better abstracting it.

WDYT? 
@marynaKK @gruebel

Reply:
let's not copy it, instead create a a new issue and add a comment to your code part to improve this part by leveraging the inheritance.